__test__/shadow-dom.spec.js

```diff
@@ -0,0 +1,186 @@
+const test = require('ava');
+const fs = require('fs');
+const path = require('path');
+
+test('Shadow DOM template files include bundler-specific conditional comments', t => {
```

Comment:
I think this is a kind of bad way for unit test.
Too much details, and it only tests the if tag, so it still passes if the end tag is missing.
It also couples unit tests with too many implementation details which will be updated relatively frequently.

Shadow-dom behaviour can be covered by the unit tests (and we absolutely need them in our setup) of the generated skeleton app itself. Those unit tests are checked by our e2e setup.

Reply:
Ah, I see. So if I add playwright tests into the generated skeletons themselves, then they get run? I actually was going to do this, but wasn't sure if shipping implementation tests like this with the generated skeletons apps the end user generates was something we wanted to do. But I am happy to implement these and agree they are absolutely essential.

Reply:
Our skeletons do provide some examples of unit tests and e2e tests. I think we provide tiny unit test for shadow-dom, but not enough to cover those variant style usages. Yes, e2e can definitely test the effects of those styles.

Maybe we just remove those tests (tailwind also maybe) for makes templates for now, and enrich our generated skeleton unit/e2e tests later.